compare/commands/compare/interactors/fetch_impacted_files.py

```diff
@@ -13,13 +22,28 @@ class ImpactedFileParameter(enum.Enum):
 
 
 class FetchImpactedFiles(BaseInteractor):
-    def _apply_filters(self, impacted_files, filters):
+    def _apply_filters(
+        self,
+        impacted_files: Optional[List[ImpactedFile]],
+        comparison: Union[PullRequestComparison, Comparison],
+        filters,
+    ):
         parameter = filters.get("ordering", {}).get("parameter")
         direction = filters.get("ordering", {}).get("direction")
         if parameter and direction:
             impacted_files = self.sort_impacted_files(
                 impacted_files, parameter, direction
             )
+        flags = filters.get("flags", [])
+        if flags and comparison:
+            head_commit_report = comparison.head_report
```

Comment:
Brought the comparison all the way down here to only compute the head_report if the flags are provided